server/jobmanager/serverv1/jobmanager_server_test.go

```diff
@@ -0,0 +1,349 @@
+		NextContext: ctx,
+	}
+	jobManager := jobmanager.NewManagerDetailed(jobmanagertest.NewMockJob, nil)
+	server := serverv1.NewJobManagerServerDetailed(jobManager)
+	req := &jobmanagerv1.StreamOutputRequest{
+		JobID:        &jobmanagerv1.JobID{Id: "not-a-valid-jobID"},
+		OutputStream: jobmanagerv1.OutputStream_STDOUT,
+	}
+
+	err := server.StreamOutput(req, mockServer)
+	assert.Error(t, err)
+}
+
+func Test_jobmanagerServer_InvalidStreamType(t *testing.T) {
+	ctx := grpcutil.AttachUserIDToContext(context.Background(), "user1")
+	mockServer := &testserverv1.MockJobmanagerStreamServer{
+		NextContext: ctx,
+	}
+	jobManager := jobmanager.NewManagerDetailed(jobmanagertest.NewMockJob, nil)
+	server := serverv1.NewJobManagerServerDetailed(jobManager)
+	req := &jobmanagerv1.StreamOutputRequest{
+		JobID:        &jobmanagerv1.JobID{Id: "1294326a-816a-4f13-8a8b-ff92c7a78984"},
+		OutputStream: jobmanagerv1.OutputStream(-1),
+	}
+
+	err := server.StreamOutput(req, mockServer)
+	assert.Error(t, err)
+}
+
+func Test_jobmanagerServer_Stream_ContextCanceled(t *testing.T) {
+	const (
+		jobName     = "myJob"
+		programPath = "/bin/ls"
+	)
+	args := []string{"-l", "/"}
+	ctx := grpcutil.AttachUserIDToContext(context.Background(), "user1")
+	ctx, cancel := context.WithCancel(ctx)
+
+	mockServer := &testserverv1.MockJobmanagerStreamServer{
+		NextContext: ctx,
+	}
+
+	jobManager := jobmanager.NewManagerDetailed(jobmanagertest.NewMockJob, nil)
+	server := serverv1.NewJobManagerServerDetailed(jobManager)
+	job, _ := server.Start(ctx, &jobmanagerv1.JobCreationRequest{
+		Name:        jobName,
+		ProgramPath: programPath,
+		Arguments:   args,
+	})
+
+	req := &jobmanagerv1.StreamOutputRequest{
+		JobID:        &jobmanagerv1.JobID{Id: job.Id.Id},
+		OutputStream: jobmanagerv1.OutputStream_STDERR,
+	}
+
+	cancel() // Intentially calling this here, not deferring it
+	err := server.StreamOutput(req, mockServer)
+
+	assert.Error(t, err)
+}
+
+func Test_jobmanagerServer_Stream_ReadSuccessfully(t *testing.T) {
+	const (
+		jobName     = "myJob"
+		programPath = "/bin/ls"
+	)
+	args := []string{"-l", "/"}
+	ctx := grpcutil.AttachUserIDToContext(context.Background(), "user1")
+	mockServer := &testserverv1.MockJobmanagerStreamServer{
+		NextContext: ctx,
+	}
+
+	jobManager := jobmanager.NewManagerDetailed(jobmanagertest.NewMockJob, nil)
+	server := serverv1.NewJobManagerServerDetailed(jobManager)
+	job, _ := server.Start(ctx, &jobmanagerv1.JobCreationRequest{
+		Name:        jobName,
+		ProgramPath: programPath,
+		Arguments:   args,
+	})
+
+	req := &jobmanagerv1.StreamOutputRequest{
+		JobID:        &jobmanagerv1.JobID{Id: job.Id.Id},
+		OutputStream: jobmanagerv1.OutputStream_STDOUT,
+	}
+
+	server.Stop(ctx, &jobmanagerv1.JobID{Id: job.Id.Id})
+	err := server.StreamOutput(req, mockServer)
+
+	assert.Nil(t, err)
+	require.NotNil(t, mockServer.LastJobOutput)
+	assert.Equal(t, []byte("this is standard output"), mockServer.LastJobOutput.Output)
```

Comment:
Perhaps you could make this a `const` in the `job_mock` so this doesn't seem so magical